app/app.go

```diff
@@ -98,6 +98,20 @@ func (a *App) InitServer() {
 			})
 		}
 		a.srv.RunJobs()
+		if a.IsLeader() {
+			a.srv.dndTask = model.CreateRecurringTaskFromNextIntervalTime("Unset DND Statuses", a.UpdateDNDStatusOfUsers, 5*time.Minute)
+		}
+		a.srv.AddClusterLeaderChangedListener(func() {
+			mlog.Info("Cluster leader changed. Determining if unset dnd status task should be running", mlog.Bool("isLeader", a.IsLeader()))
+			if a.IsLeader() {
```

Comment:
This whole event happens inside a goroutine. We need to guard access to `dndTask` with a mutex.